src/pages/api/auth-connect/re-send-identity-validation-code.ts

```diff
@@ -0,0 +1,152 @@
+import { withAuth } from "@src/middlewares/with-auth";
+import { withSentry } from "@src/middlewares/with-sentry";
+import { getDBUserFromSub } from "@src/queries/get-db-user-from-sub";
+import { getIdentityType } from "@src/utils/get-identity-type";
+
+const handler: Handler = (request, response): Promise<void> => {
+  return getTracer().span(
+    "send-identity-validation-code handler",
+    async (span) => {
+      if (!request.body.eventId) {
+        // WebError
+        throw new Error();
+      }
+
+      const userCookie = await getServerSideCookies<UserCookie>(request, {
+        cookieName: "user-cookie",
+        isCookieSealed: true,
+        cookieSalt: config.cookieSalt,
+      });
+
+      if (userCookie) {
+        const user = await getDBUserFromSub(userCookie.sub);
+
+        if (!user?.temporary_identities) {
+          // WebError
+          throw new Error();
+        }
+
+        const temporaryIdentity = user.temporary_identities.find(
+          ({ eventId }) => eventId === request.body.eventId,
+        );
+
+        if (!temporaryIdentity) {
+          // WebError
+          throw new Error();
+        }
+
+        const {
+          type,
+          value,
+          callbackUrl,
+          expiresAt,
+          primary,
+        } = temporaryIdentity;
+
+        const identity = {
+          type: getIdentityType(type),
+          value: value,
+        };
+
+        return await sendIdentityValidationCode(config.managementCredentials, {
+          callbackUrl,
+          identity,
+          localeCodeOverride: "en-EN",
+          userId: userCookie.sub,
+        })
+          .then(async ({ eventId }) => {
+            span.setDisclosedAttribute("is validation code sent", true);
+
+            const temporaryIdentity = {
+              eventId,
+              value,
+              type,
+              expiresAt,
+              primary,
+              callbackUrl,
+            };
+
+            await insertTemporaryIdentity(userCookie.sub, temporaryIdentity);
+
+            const verificationCodeMessage =
+              getIdentityType(type) === IdentityTypes.EMAIL
+                ? "Confirmation email has been sent"
+                : "Confirmation SMS has been sent";
+
+            setAlertMessagesCookie(response, verificationCodeMessage);
+
+            response.statusCode = HttpStatus.OK;
+            response.setHeader("Content-Type", "application/json");
+            response.json({ eventId });
+            return;
+          })
+          .catch((error) => {
+            if (
+              error instanceof IdentityAlreadyUsedError ||
+              error instanceof IdentityValueCantBeBlankError
+            ) {
+              span.setDisclosedAttribute(
+                "Identity input exception",
+                error.message,
+              );
+
+              response.statusCode = HttpStatus.BAD_REQUEST;
+              response.json({ errorMessage: error.message });
+              return;
```

Comment:
`// WebError!` 😇

Reply:
Should I go with a simple
```ts
  BAD_REQUEST: {
    code: "bad_request",
    httpStatus: HttpStatus.BAD_REQUEST,
    message: "Request malformed",
  },
```
or specify that it related to bad input?

Reply:
Well, in the case of a blank value, I think the error could be explicit.
In the case of an already used error, could you make it a generic bad request for now and make a point to talk about it on the next sprint meeting? I don't know what's our stance for this error since it could be used to check that an email is present in our db.

Reply:
> that an email is present in our db.
Omg, good point x_x.